src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/unit/service/cas1/Cas1SpaceBookingManagementDomainEventServiceTest.kt

```diff
@@ -0,0 +1,218 @@
+      assertThat(data.previousExpectedDepartureOn).isNull()
+      assertThat(data.applicationId).isEqualTo(application.id)
+      assertThat(data.applicationSubmittedOn).isEqualTo(application.submittedAt!!.toLocalDate())
+      assertThat(data.applicationUrl).isEqualTo("http://frontend/applications/${application.id}")
+      assertThat(data.arrivedAt).isEqualTo(arrivalDate)
+      assertThat(data.deliusEventNumber).isEqualTo(application.eventNumber)
+      assertThat(data.premises.id).isEqualTo(premises.id)
+      assertThat(data.premises.name).isEqualTo(premises.name)
+      assertThat(data.premises.apCode).isEqualTo(premises.apCode)
+      assertThat(data.premises.legacyApCode).isEqualTo(premises.qCode)
+      assertThat(data.premises.localAuthorityAreaName).isEqualTo(premises.localAuthorityArea!!.name)
+    }
+
+    @Test
+    fun `record arrival and emits domain event recognising change in expected departure date`() {
+      val previousExpectedDepartureDate = departureDate.plusMonths(1)
+      service.arrivalRecorded(existingSpaceBooking, previousExpectedDepartureDate)
+
+      val domainEventArgument = slot<DomainEvent<PersonArrivedEnvelope>>()
+
+      verify(exactly = 1) {
+        domainEventService.savePersonArrivedEvent(
+          capture(domainEventArgument),
+          emit = true,
+        )
+      }
+
+      val domainEvent = domainEventArgument.captured
+
+      assertThat(domainEvent.data.eventType).isEqualTo(EventType.personArrived)
+      assertThat(domainEvent.data.timestamp).isWithinTheLastMinute()
+      assertThat(domainEvent.applicationId).isEqualTo(application.id)
+      assertThat(domainEvent.bookingId).isNull()
+      assertThat(domainEvent.cas1SpaceBookingId).isEqualTo(existingSpaceBooking.id)
+      assertThat(domainEvent.crn).isEqualTo(existingSpaceBooking.crn)
+      assertThat(domainEvent.nomsNumber).isEqualTo(caseSummary.nomsId)
+      assertThat(domainEvent.occurredAt).isEqualTo(existingSpaceBooking.actualArrivalDateTime)
+      val data = domainEvent.data.eventDetails
+      assertThat(data.previousExpectedDepartureOn).isEqualTo(previousExpectedDepartureDate)
+      assertThat(data.expectedDepartureOn).isEqualTo(departureDate)
+      assertThat(data.applicationId).isEqualTo(application.id)
+      assertThat(data.applicationSubmittedOn).isEqualTo(application.submittedAt!!.toLocalDate())
+      assertThat(data.applicationUrl).isEqualTo("http://frontend/applications/${application.id}")
+      assertThat(data.arrivedAt).isEqualTo(arrivalDate)
+      assertThat(data.deliusEventNumber).isEqualTo(application.eventNumber)
+      assertThat(data.premises.id).isEqualTo(premises.id)
+      assertThat(data.premises.name).isEqualTo(premises.name)
+      assertThat(data.premises.apCode).isEqualTo(premises.apCode)
+      assertThat(data.premises.legacyApCode).isEqualTo(premises.qCode)
+      assertThat(data.premises.localAuthorityAreaName).isEqualTo(premises.localAuthorityArea!!.name)
```

Comment:
i would be included to not repeat all these assertions and just focus on the field that is different from other unit tests